eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -0,0 +1,324 @@
+    val mainPenaltyTxCodec: Codec[MainPenaltyTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[MainPenaltyTx]
+    val htlcPenaltyTxCodec: Codec[HtlcPenaltyTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[HtlcPenaltyTx]
+    val claimHtlcDelayedOutputPenaltyTxCodec: Codec[ClaimHtlcDelayedOutputPenaltyTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimHtlcDelayedOutputPenaltyTx]
+    val claimLocalAnchorOutputTxCodec: Codec[ClaimLocalAnchorOutputTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimLocalAnchorOutputTx]
+    val claimRemoteAnchorOutputTxCodec: Codec[ClaimRemoteAnchorOutputTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimRemoteAnchorOutputTx]
+    val closingTxCodec: Codec[ClosingTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec) :: ("outputIndex" | optional(bool8, outputInfoCodec))).as[ClosingTx]
+
+    val txWithInputInfoCodec: Codec[TransactionWithInputInfo] = discriminated[TransactionWithInputInfo].by(uint16)
+      .typecase(0x01, commitTxCodec)
+      .typecase(0x02, htlcSuccessTxCodec)
+      .typecase(0x03, htlcTimeoutTxCodec)
+      .typecase(0x04, claimHtlcSuccessTxCodec)
+      .typecase(0x05, claimHtlcTimeoutTxCodec)
+      .typecase(0x06, claimP2WPKHOutputTxCodec)
+      .typecase(0x07, claimLocalDelayedOutputTxCodec)
+      .typecase(0x08, mainPenaltyTxCodec)
+      .typecase(0x09, htlcPenaltyTxCodec)
+      .typecase(0x10, closingTxCodec)
+      .typecase(0x11, claimLocalAnchorOutputTxCodec)
+      .typecase(0x12, claimRemoteAnchorOutputTxCodec)
+      .typecase(0x13, claimRemoteDelayedOutputTxCodec)
+      .typecase(0x14, claimHtlcDelayedOutputPenaltyTxCodec)
+      .typecase(0x15, htlcDelayedTxCodec)
+
+    val claimRemoteCommitMainOutputTxCodec: Codec[ClaimRemoteCommitMainOutputTx] = discriminated[ClaimRemoteCommitMainOutputTx].by(uint8)
+      .typecase(0x01, claimP2WPKHOutputTxCodec)
+      .typecase(0x02, claimRemoteDelayedOutputTxCodec)
+
+    val claimAnchorOutputTxCodec: Codec[ClaimAnchorOutputTx] = discriminated[ClaimAnchorOutputTx].by(uint8)
+      .typecase(0x01, claimLocalAnchorOutputTxCodec)
+      .typecase(0x02, claimRemoteAnchorOutputTxCodec)
+
+    val htlcTxCodec: Codec[HtlcTx] = discriminated[HtlcTx].by(uint8)
+      .typecase(0x01, htlcSuccessTxCodec)
+      .typecase(0x02, htlcTimeoutTxCodec)
+
+    val claimHtlcTxCodec: Codec[ClaimHtlcTx] = discriminated[ClaimHtlcTx].by(uint8)
+      .typecase(0x01, claimHtlcSuccessTxCodec)
+      .typecase(0x02, claimHtlcTimeoutTxCodec)
+
+    val htlcTxsAndRemoteSigsCodec: Codec[HtlcTxAndRemoteSig] = (
+      ("txinfo" | htlcTxCodec) ::
+        ("remoteSig" | lengthDelimited(bytes64))).as[HtlcTxAndRemoteSig]
+
+    val commitTxAndRemoteSigCodec: Codec[CommitTxAndRemoteSig] = (
+      ("commitTx" | commitTxCodec) ::
+        ("remoteSig" | lengthDelimited(bytes64))).as[CommitTxAndRemoteSig]
```

Comment:
nit: same here, we could remove the `lengthDelimited`

Reply:
I'm not completely sure because we have been bitten when we tried to be too smart. This seems harmless, but I wonder why we didn't do it during the 0->1 transition.

That comment is odd:
```
// we store as variable length for historical purposes (we used to store as DER encoded)
```

Reply:
It's true that this is quite harmless since it's only 2 bytes. And thinking about this more, if later in the future we have potentially different signature sizes (schnorr?) we'd be happy that we length-prefixed it. So let's keep it like that for now.

Reply:
To be honest, if we introduce a new type of signature we will probably prefix them by a tag. I'm tempted to go ahead and remove that length prefix

Reply:
That's true, we would likely need a change anyway when we introduce a different signature format.
It's your call then, we can remove that!

Reply:
Done in 2e074f710 (squashed).